### ISSUE TITLE
🚧 Link List Gizmo: A better way to display a marketplace?

### ISSUE THREAD

**Issue by rosschapman** [zinc-collective/convene#2036]
In Sutty's [recent heuristic review](https://github.com/zinc-collective/convene/issues/1980) of the Convene Marketplace they noted that Piikup's vendor list was lacked important visual cues like images and stronger buttons. 

While [Adding images](https://github.com/zinc-collective/convene/issues/1961) and allowing button configurations to Sections would be a good start to improving the user experience, the feedback is good cause to consider that the `Section Navigation` gizmo does not allow enough customization for different _home page_ use cases. 

For example, currently the Section Navigation gizmo does not allow: 
- Re-ordering sections

Might _link list_ gizmo that allows full customization of title, image, button, url, card type (new concept) be a better solution for marketplace hosts or similar use cases?

**Comment by zspencer** [comment]
I did complete the migration of removing the old "every `Section` supports navigation to every other `Section`" functionality and added the `SectionNavigation` to the `Entrance` for https://marketplace.piikup.com. I don't think I did it anywhere else tho 🙈.

Re: `LinkList` deprecates `SectionNavigation` - I think both are useful; as  the `SectionNavigation`  automatically linking to all available `Sections` is a pretty strong win; and being able to link to anything is a pretty strong win. I don't think we should merge them. Also +1 to @anaulin's point that `LinkList` gizmo would ideally allow a Convene `Space` to compete directly with LinkTree and similar sites. (I would also love for us to be in a position where we had "too many gizmos" tho, so take this with an iceberg of salt)

Re: Improving Visuals - I was thinking that a neat way to do a standards-forward step would be adding [opengraph](https://ogp.me) support to `Sections`. `Members` could set the visuals and text the OpenGraph would show, and then when anyone links to a `Section` it will look fantastic. Once that worked, we would update the `SectionNavigation` (or the `LinkNavigation`) to consume the OpenGraph data. To make the `LinkList` or `SectionNavigation` more robust, we could update them to allow that data to be over-written manually by a `Member` or even refresh it periodically.


TL/DR - I would recommend:

- Adding [`OpenGraph` data to `Section`](https://github.com/zinc-collective/convene/issues/1961)
- Consuming the `OpenGraph` data in the existing `SectionNavigation`
- Following up eventually with a new `LinkList` gizmo; especially since Instagram's "You get ONE URL. ONE" policy ain't going anywhere soon.

**Comment by anaulin** [comment]
I am not sure that the images a user will want to add to a link to a section are the same as the ones they'll want to have displayed in a "preview card" or anywhere else. I also don't think that we should limit our users to at most one image per link, which is what OpenGraph dictates.

I think it would be better to focus first on building and defining the functionality we want in a configurable "list of links" or "list of sections". If any of the data we end up collecting as part of that ends up fitting the OpenGraph or Schema.org or any of the various other formats for this, then we can expose it in those formats as well pretty easily.

**Comment by zspencer** [comment]
So if we want to get people to want to buy some noms from the `Vendors` in the Piikup `Marketplace`, I could see the navigation looking like:
- A mouth-watering image that looks decent when at full-width on a mobile phone (i.e. https://www.crumbleandwhisk.com/farmers-markets)
- Alt text describing that image
- The name of the restaurant
- Some flavor text describing the Restaurant that really highlights why it's a great place to get food
- Maybe high-value context like "Catering" or "Groceries" or "Gluten Free" or "Thai" or "Desserts"?

(I am now imagining a world where what we actually work on next is the `Marketplace::LandingPage` that is even more specific than this or `SectionNavigation` and also helps us solve the "Probably should have more context on the landing page" problem)

**Comment by anaulin** [comment]
Yeah, I've also been mulling over the "context for the marketplace" piece. I think we could do it adding an [ActionText](https://guides.rubyonrails.org/action_text_overview.html) field, editable from the Marketplace configuration.

It would be neat to have a Space-wide "image library", so when you are doing anything that requires an image, like, say, you are configuring a section link, you can choose from a gallery with all the images that have been uploaded to that Space, including, say, all the product images in the contained marketplaces. Not sure if that would scale well, though, and it is certainly not a "must have".

All that said, we gotta start somewhere, and starting with making the list of section links more configurable (and to stop linking to the current page forcatssake) is a good place to start.

**Comment by rosschapman** [comment]
**💡 _LinkList with Internal Links_**

If our Convene `LinkList` Gizmo has a toggle on the "url field" to show "internal links" and allow setting a link to an "internal destination" via a autocomplete dropdown of Sections or other other content types, that would also make it worthwhile over a third-party integration.

**Comment by rosschapman** [comment]
> It would be neat to have a Space-wide "image library", so when you are doing anything that requires an image, like, say, you are configuring a section link, you can choose from a gallery with all the images that have been uploaded to that Space, including, say, all the product images in the contained marketplaces. 

💯 

> Not sure if that would scale well, though, and it is certainly not a "must have".

What aspect of scaling are you worried about? Cost to us?

**Comment by zspencer** [comment]
Re: Space-wide Image Library - I would love for us to treat Files as an OS-level thing; We definitely should segment files by  by `Space`; and at some point we'll want to start measuring storage + xfer (and CDN).

Re: Internal inks - I'm imagining something where when you add a link you can search all Linkable Resources with a fuzzy-finder or just copy-paste a URL. 

This way, you could easily link to say, a `Section` or a`Marketplace::Product` or `Journal::Entry`; and everything looks spiffy.

While I think the idea of say, auto-adding `Sections` is interesting, I think a single-purpose Gizmo for `SectionNavigation` is a better solution. Trying to make two things one things is perhaps as much of a drag-risk as trying to make one thing two things.

**Comment by zspencer** [comment]
P.s. I've added some updated use cases to https://github.com/zinc-collective/convene/issues/1988 based upon this conversation. I still think an independent Link List is valuable; but I don't want to lose track of the ideas that are more applicable to the `SectionNavigation` gizmo

**Comment by anaulin** [comment]
>  Not sure if that would scale well, though, and it is certainly not a "must have".
> @anaulin What aspect of scaling are you worried about? Storage cost? User interface complicatedness?

I am more worried about the UX. It seems like you could quickly end up with more than a dozen of images in a Space, which would make finding the one you want obnoxious. Although perhaps that could be mitigated by e.g. showing the images in most-recent-first order. It just seems non-trivial

**Comment by rosschapman** [comment]
I use MarsEdit on mac for blogging and it has a very simple media manager that lists your images ordered by upload date with a single search input that appears to fuzzy match only on filename. As an MVP something like this would be pretty solid methinks. 

![image](https://github.com/zinc-collective/convene/assets/5185/e2b5d1a0-b486-4a4e-a196-3fce15a43f5b)
